ZelBack/src/services/appsService.js

```diff
@@ -8933,92 +8940,89 @@ async function checkMyAppsAvailability() {
-      // if error, following request will fail increasing dos too
-      if (myAppAvailability && myAppAvailability.data.status === 'success') {
-        if (myAppAvailability.data.data) { // this is a data response containing array of ports listening
-          data = { // adjust data to only check for listening ports
-            ip: myIP,
-            appname: app.name,
-            ports: myAppAvailability.data.data,
-            pubKey,
-          };
-          const secondStringData = JSON.stringify(data);
-          // eslint-disable-next-line no-await-in-loop
-          const secondSignature = await signCheckAppData(secondStringData);
-          data.signature = secondSignature;
-          // eslint-disable-next-line no-await-in-loop
-          const resMyAppAvailability = await axios.post(`http://${askingIP}:${askingIpPort}/flux/checkappavailability`, JSON.stringify(data), axiosConfig).catch((error) => {
-            log.error(`${askingIP} for app availability is not reachable`);
-            log.error(error);
-          });
-          if (resMyAppAvailability && resMyAppAvailability.data.status === 'error') {
-            log.warn(`Running application ${app.name} on ports ${JSON.stringify(appPorts)}, unavailability detected from ${askingIP}:${askingIpPort}`);
-            log.warn(JSON.stringify(data));
-            currentDos += 0.4;
-            dosState += 0.4;
-          } else if (resMyAppAvailability && resMyAppAvailability.data.status === 'success') {
-            log.info(`${resMyAppAvailability.data.data.message} Detected from ${askingIP}:${askingIpPort}`);
-          }
-        } else {
-          log.info(`Running application ${app.name} on ports ${JSON.stringify(appPorts)} open but app not listening at them.`);
-        }
-      } else if (myAppAvailability && myAppAvailability.data.status === 'error') {
-        log.warn(`Running application ${app.name} on ports ${JSON.stringify(appPorts)}, unavailability detected from inside`);
-        log.warn(JSON.stringify(data));
-        currentDos += 0.4;
-        dosState += 0.4;
-      }
+    }
+    // choose random port
+    const min = config.fluxapps.portMin - 1000;
+    const max = config.fluxapps.portMax;
+    testingPort = Math.floor(Math.random() * (max - min) + min);
+    if (appPorts.includes(testingPort)) {
+      // skip this check
+      await serviceHelper.delay(60 * 60 * 1000);
+      checkMyAppsAvailability();
+      return;
+    }
+    // now open this port properly and launch listening on it
+    await fluxNetworkHelper.allowPort(testingPort);
+    if ((userconfig.initial.apiport && userconfig.initial.apiport !== config.server.apiport) || isUPNP) {
+      await upnpService.mapUpnpPort(testingPort, 'Flux_Test_App');
+    }
+    await serviceHelper.delay(5 * 60 * 1000);
+    testingAppserver.listen(testingPort);
+    await serviceHelper.delay(10 * 60 * 1000);
+    // eslint-disable-next-line no-await-in-loop
```

Comment:
5 + 10m of service delay here. Isn't that too much time?

Reply:
Oh good catch. Was meant to be 5 seconds, 10 seconds